jwplayer-appletv-web-app/js/playback/Playback.js

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright 2015 Longtail Ad Solutions Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ **/
+
+/**
+ * The Playback module is a wrapper around Apple's TV JS Player class.
+ * In addition to the TV JS Player, the Playback module broadcasts certain events during
+ * set up and playback.
+ */
+var Playback = (function() {
+
+  var player = new Player();
+  player.addEventListener("stateDidChange", stateHandler);
+  player.addEventListener("timeDidChange", timeHandler, {
+    interval: 1
+  });
+  player.addEventListener("mediaItemWillChange", mediaItemWillChangeHandler);
+  player.addEventListener("mediaItemDidChange", mediaItemChangedHandler);
+  player.addEventListener("requestSeekToTime", requestSeekToTimeHandler);
+  player.addEventListener("timedMetadata", timedMetadataHandler);
+  player.addEventListener("shouldHandleStateChange", shouldHandleStateChangeHandler);
+
+  function load(playlist) {
+    player.playlist = playlist;
+    player.addEventListener("timeBoundaryDidCross", timeHandler, [playlist.item(0).duration]);
+    EventBus.publish(Events.PLAYLIST_LOADED, {
+      playlist: playlist
+    });
+  }
+
+  function play() {
+    player.play();
+  }
+
+  function seek() {
+    player.seek();
+  }
+
+  function pause() {
+    EventBus.publish(Events.PAUSE);
+    player.pause();
+  }
+
+  function stop() {
+    player.stop();
+  }
+
+  function present() {
+    player.present();
+  }
+
+  function setOverlay(overlayDocument) {
+    EventBus.publish(Events.PLAYER_OVERLAY, {
+      overlayDocument: overlayDocument
+    });
+    player.overlayDocument = overlayDocument;
+  }
+
+  /**
+   * Handles player stateDidChange event
+   */
+  function stateHandler(evt) {
+    //console.log('stateHandler %O', evt);
+    EventBus.publish(Events.PLAYER_STATE, {
+      state: evt.state,
```

Comment:
I wonder if it might make things a bit more legible if we created our own enum for potential player states?  Apple doesn't even [document](https://developer.apple.com/library/tvos/documentation/TVMLJS/Reference/TVJSPlayer_Ref/index.html#//apple_ref/javascript/instp/Player/stateDidChange) what values the `stateDidChange.state` can have.